File: code/community/1306402371709964289/deepgram-diarization-pricing.cs
using System;
using System.Net.Http;
using System.Text;
using System.Threading.Tasks;
using Microsoft.Extensions.Configuration;

class Program
{
    private static readonly HttpClient client = new HttpClient();

    static async Task Main()
    {
        var builder = new ConfigurationBuilder()
            .AddEnvironmentVariables();
        var configuration = builder.Build();

        string apiKey = configuration["DEEPGRAM_API_KEY"];
        string audioUrl = configuration["AUDIO_URL"];

        var requestContent = new StringContent(
            "{"url":"" + audioUrl + "","diarize":true}",
            Encoding.UTF8,
            "application/json"
        );

        client.DefaultRequestHeaders.Add("Authorization", "Token " + apiKey);

        var response = await client.PostAsync("https://api.deepgram.com/v1/listen", requestContent);
        var responseString = await response.Content.ReadAsStringAsync();

        Console.WriteLine(responseString);
    }
}

File: code/community/1306330637468176395/deepgram-diarization-pricing.go
package main

import (
    "bytes"
    "encoding/json"
    "fmt"
    "io/ioutil"
    "net/http"
    "os"
)

func main() {
    apiKey := os.Getenv("DEEPGRAM_API_KEY")
    audioUrl := os.Getenv("AUDIO_URL")

    payload := map[string]interface{}{
        "url": audioUrl,
        "diarize": true,
    }

    jsonPayload, _ := json.Marshal(payload)

    req, err := http.NewRequest("POST", "https://api.deepgram.com/v1/listen", bytes.NewBuffer(jsonPayload))
    if err != nil {
        fmt.Println("Error creating request:", err)
        return
    }

    req.Header.Set("Authorization", fmt.Sprintf("Token %s", apiKey))
    req.Header.Set("Content-Type", "application/json")

    client := &http.Client{}
    resp, err := client.Do(req)
    if err != nil {
        fmt.Println("Error making request:", err)
        return
    }

    defer resp.Body.Close()
    body, _ := ioutil.ReadAll(resp.Body)

    fmt.Println("Transcription result:", string(body))
}